cas-server-webapp-actions-aup-webflow/src/main/java/org/jasig/cas/web/flow/DefaultAcceptableUsagePolicyRepository.java

```diff
@@ -0,0 +1,37 @@
+package org.jasig.cas.web.flow;
+
+import org.jasig.cas.authentication.Credential;
+import org.springframework.stereotype.Component;
+import org.springframework.webflow.execution.RequestContext;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * This is {@link DefaultAcceptableUsagePolicyRepository}.
+ *
+ * @author Misagh Moayyed
+ * @since 4.2
+ */
+@Component("defaultAcceptableUsagePolicyRepository")
+public class DefaultAcceptableUsagePolicyRepository implements AcceptableUsagePolicyRepository {
+    private static final long serialVersionUID = -3059445754626980894L;
+
+    private final Map<String, Boolean> policyMap = new ConcurrentHashMap<>();
```

Comment:
Even though the concurrent hash map is used, the impl might not be thread safe. Might need to synchronize on this field in verify and submit methods to promote a consistent visibility between reader and writer threads.